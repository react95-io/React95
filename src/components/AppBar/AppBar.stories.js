import React from 'react';

import {
  AppBar,
  Toolbar,
  TextField,
  Button,
  List,
  ListItem,
  Divider
} from 'react95';

import LogoIcon from '../Icon/LogoIcon';

export default {
  title: 'AppBar',
  component: AppBar,
  decorators: [
    story => (
      <div
        style={{
          padding: '5rem',
          background: 'teal'
        }}
      >
        {story()}
      </div>
    )
  ]
};

export const Default = () => {
  const [open, setOpen] = React.useState(false);

  return (
    <AppBar>
      <Toolbar style={{ justifyContent: 'space-between' }}>
        <div style={{ position: 'relative', display: 'inline-block' }}>
          <Button
            onClick={() => setOpen(!open)}
            active={open}
            style={{ fontWeight: 'bold' }}
          >
            <LogoIcon style={{ marginLeft: -2, marginRight: 4 }} />
            Start
          </Button>
          {open && (
            <List
              style={{
                position: 'absolute',
                left: '0',
                top: '100%'
              }}
              onClick={() => setOpen(false)}
            >
              <ListItem>
                <span role='img' aria-label='👨‍💻'>
                  👨‍💻
                </span>
                Profile
              </ListItem>
              <ListItem>
                <span role='img' aria-label='📁'>
                  📁
                </span>
                My account
              </ListItem>
              <Divider />
              <ListItem disabled>
                <span role='img' aria-label='🔙'>
                  🔙
                </span>
                Logout
              </ListItem>
            </List>
          )}
        </div>

        <TextField placeholder='Search...' width={150} />
      </Toolbar>
    </AppBar>
  );
};

Default.story = {
  name: 'default'
};
