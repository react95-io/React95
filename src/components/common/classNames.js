/**
 * Given a string, number, object, or array, extract
 * the class names configuration from it, and return
 * as a single class string.
 * @param {*} [value]
 * @return {string}
 */
export const getClassesForValue = value => {
  if (value) {
    if (typeof value === 'string' || typeof value === 'number') {
      return value.toString();
    }

    if (Array.isArray(value)) {
      return value.map(getClassesForValue).join(' ');
    }

    if (typeof value === 'object' && !Array.isArray(value)) {
      return Object.keys(value)
        .filter(key => Boolean(value[key]))
        .map(getClassesForValue)
        .join(' ');
    }
  }

  return '';
};

/**
 * Given any number of arguments of type string, number,
 * object, or a multidimensional array, extract a class names
 * configuration from it, and return as a single classes string.
 * For object values, uses keys as classes, but only if their
 * associated values are truthy.
 * @example
 * const result = classNames('alpaca', ['bear', 'cat']);
 * --> 'alpaca bear cat'
 * const result = classNames('alpaca', { bear: null, cat: true });
 * --> 'alpaca cat' // 'bear' is null, so is not included
 * const result = classNames(['alpaca', { bear: true }, ['cat']]);
 * --> 'alpaca bear cat' // multi-dim array is flattened
 * @param {...*} args
 * @return {string}
 */
const classNames = (...args) => {
  return args.reduce((classesString, arg) => {
    const classesForArg = getClassesForValue(arg);

    // leading whitespace is okay as the final string is trimmed
    return `${classesString} ${classesForArg}`.trim();
  }, '');
};

export default classNames;
