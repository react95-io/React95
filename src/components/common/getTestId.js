const getTestId = (testIdPrefix, appendText = '') =>
  testIdPrefix ? `${testIdPrefix}${appendText}`.trim() : undefined;

export default getTestId;
