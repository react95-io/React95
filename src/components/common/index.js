import { css } from "styled-components";

export const shadow = `4px 4px 10px 0 rgba(0,0,0,0.35)`;
export const insetShadow = `inset 3px 3px 10px rgba(0, 0, 0, 0.2)`;

export const createDisabledTextStyles = () => css`
  color: ${({ theme }) => theme.textDisabled};
  text-shadow: 1px 1px ${({ theme }) => theme.textDisabledShadow};
  /* filter: grayscale(100%); */
`;
export const createBoxStyles = () =>
  css`
    box-sizing: border-box;
    display: inline-block;
    background-color: ${({ theme }) => theme.material};
    color: ${({ theme }) => theme.text};
  `;
export const createBorderStyles = (invert = false) =>
  invert
    ? css`
        border-style: solid;
        border-width: 2px;
        border-left-color: ${({ theme }) => theme.borderDarkest};
        border-top-color: ${({ theme }) => theme.borderDarkest};
        border-right-color: ${({ theme }) => theme.borderLightest};
        border-bottom-color: ${({ theme }) => theme.borderLightest};
        box-shadow: ${props => props.shadow && shadow + ", "} inset 1px 1px 0px
            1px ${({ theme }) => theme.borderDark},
          inset -1px -1px 0 1px ${({ theme }) => theme.borderLight};
      `
    : css`
        border-style: solid;
        border-width: 2px;
        border-left-color: ${({ theme }) => theme.borderLightest};
        border-top-color: ${({ theme }) => theme.borderLightest};
        border-right-color: ${({ theme }) => theme.borderDarkest};
        border-bottom-color: ${({ theme }) => theme.borderDarkest};
        box-shadow: ${props => props.shadow && shadow + ", "} inset 1px 1px 0px
            1px ${({ theme }) => theme.borderLight},
          inset -1px -1px 0 1px ${({ theme }) => theme.borderDark};
      `;
