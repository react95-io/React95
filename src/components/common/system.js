// TODO - implement styled-system

export const blockSizes = {
  sm: '27px',
  md: '35px',
  lg: '43px'
};

export const fontFamily = 'sans-serif';
