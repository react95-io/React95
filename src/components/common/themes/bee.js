export default {
  name: 'bee',

  anchor: '#1034a6',
  anchorVisited: '#440381',
  borderDark: '#846d06',
  borderDarkest: '#0C1618',
  borderLight: '#e7c221',
  borderLightest: '#f8df6e',
  canvas: '#ffffff',
  canvasText: '#0C1618',
  canvasTextDisabled: '#846d06',
  canvasTextDisabledShadow: '#f8df6e',
  canvasTextInvert: '#ffffff',
  checkmark: '#0C1618',
  checkmarkDisabled: '#846d06',
  flatDark: '#9e9e9e',
  flatLight: '#d8d8d8',
  focusSecondary: '#fefe03',
  headerBackground: '#0C1618',
  headerNotActiveBackground: '#7F7B82',
  headerNotActiveText: '#e5bd03',
  headerText: '#f8df6e',
  hoverBackground: '#0C1618',
  material: '#e5bd03',
  materialDark: '#7F7B82',
  materialText: '#0C1618',
  materialTextDisabled: '#846d06',
  materialTextDisabledShadow: '#f8df6e',
  materialTextInvert: '#ffffff',
  progress: '#0C1618',
  tooltip: '#fefbcc'
};
