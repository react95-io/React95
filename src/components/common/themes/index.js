import azureOrange from './azureOrange';
import bee from './bee';
import blackAndWhite from './blackAndWhite';
import brick from './brick';
import candy from './candy';
import coldGray from './coldGray';
import eggplant from './eggplant';
import highContrast from './highContrast';
import lilac from './lilac';
import lilacRoseDark from './lilacRoseDark';
import maple from './maple';
import marine from './marine';
import matrix from './matrix';
import millenium from './millenium';
import modernDark from './modernDark';
import molecule from './molecule';
import ninjaTurtles from './ninjaTurtles';
import olive from './olive';
import original from './original';
import pamelaAnderson from './pamelaAnderson';
import plum from './plum';
import rainyDay from './rainyDay';
import rose from './rose';
import slate from './slate';
import spruce from './spruce';
import theSixtiesUSA from './theSixtiesUSA';
import tokyoDark from './tokyoDark';
import tooSexy from './tooSexy';
import travel from './travel';
import vaporTeal from './vaporTeal';
import vermillion from './vermillion';
import violetDark from './violetDark';
import water from './water';

export default {
  azureOrange,
  bee,
  blackAndWhite,
  brick,
  candy,
  coldGray,
  eggplant,
  highContrast,
  lilac,
  lilacRoseDark,
  maple,
  marine,
  matrix,
  millenium,
  modernDark,
  molecule,
  ninjaTurtles,
  olive,
  original,
  pamelaAnderson,
  plum,
  rainyDay,
  rose,
  slate,
  spruce,
  theSixtiesUSA,
  tokyoDark,
  tooSexy,
  travel,
  vaporTeal,
  vermillion,
  violetDark,
  water
};
