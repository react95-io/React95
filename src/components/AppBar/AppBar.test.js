import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PropTypes from "prop-types";
import { assertPropTypes } from "check-prop-types";

import AppBar from "./AppBar";
import themes from "../common/themes";
import { shadow } from "../common";

describe("Component: AppBar", () => {

  const testProps = {
    fixed: true,
    className: "my-class",
    style: { color: "red" },
    children: "foo",
    shadow: true,
  }

  it("warns if props are missing or the wrong type", () => {
    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, fixed: 123 }, 'prop', AppBar.fixed); }).toThrow();
    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, shadow: 123 }, 'prop', AppBar.fixed); }).toThrow();
    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, children: undefined }, 'prop', AppBar.children); }).toThrow();
    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, className: 123 }, 'prop', AppBar.className); }).toThrow();
    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, style: "red" }, 'prop', AppBar.style); }).toThrow();
  })


  it("renders", () => {
    const wrapper = shallow(<AppBar />);
    expect(wrapper.find("AppBar__StyledAppBar").length).toBe(1);
  });

  it("renders children", () => {
    const wrapper = shallow(<AppBar>text</AppBar>);
    expect(wrapper.text()).toBe("text");
  });

  it("renders className", () => {
    const wrapper = shallow(<AppBar className="my-class">text</AppBar>);
    expect(wrapper.props().className).toBe("my-class");
  });

  it("renders style object", () => {
    const wrapper = shallow(
      <AppBar
        href="#"
        className="my-class"
        style={{ color: "blue" }}
      >
        text
      </AppBar>
    );
    expect(wrapper.props().style.color).toBe("blue");
  })

  it("renders other props passed", () => {
    const wrapper = shallow(
      <AppBar
        href="#"
        className="my-class"
        style={{ color: "blue" }}
        a={1}
        b={2}
        c={3}
      >
        text
      </AppBar>
    );
    expect(wrapper.props().a).toBe(1);
    expect(wrapper.props().b).toBe(2);
    expect(wrapper.props().c).toBe(3);
  });

  it("renders default props", () => {
    const wrapper = shallow(<AppBar />);
    expect(wrapper.props().shadow).toBe(true);
    expect(wrapper.props().fixed).toBe(true);
  });

  it("renders default component styles", () => {
    const tree = renderer.create(<AppBar theme={themes.default} />).toJSON();
    expect(tree).toHaveStyleRule("box-sizing", "border-box");
    expect(tree).toHaveStyleRule("position", "fixed");
    expect(tree).toHaveStyleRule("background-color", themes.default.material);
    expect(tree).toHaveStyleRule("color", themes.default.text);
    expect(tree).toHaveStyleRule("border-style", "solid");
    expect(tree).toHaveStyleRule("border-width", "2px");
    expect(tree).toHaveStyleRule("border-left-color", themes.default.borderLightest);
    expect(tree).toHaveStyleRule("border-top-color", themes.default.borderLightest);
    expect(tree).toHaveStyleRule("border-right-color", themes.default.borderDarkest);
    expect(tree).toHaveStyleRule("border-bottom-color", themes.default.borderDarkest);
    expect(tree).toHaveStyleRule("box-shadow", `${shadow},inset 1px 1px 0px 1px ${themes.default.borderLight},inset -1px -1px 0 1px ${themes.default.borderDark}`);
    expect(tree).toHaveStyleRule("top", "0");
    expect(tree).toHaveStyleRule("right", "0");
    expect(tree).toHaveStyleRule("left", "auto");
    expect(tree).toHaveStyleRule("display", "flex");
    expect(tree).toHaveStyleRule("flex-direction", "column");
    expect(tree).toHaveStyleRule("width", "100%");
  });

  it("renders position absolute", () => {
    const tree = renderer.create(<AppBar fixed={false} />).toJSON();
    expect(tree).toHaveStyleRule("position", "absolute");
  });

  it("render matching snapshot", () => {
    const component = renderer.create(
      <AppBar>
        test
      </AppBar>
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
