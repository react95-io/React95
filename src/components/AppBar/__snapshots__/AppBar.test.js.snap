// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AppBar render matching snapshot 1`] = `
.c0 {
  border-style: solid;
  border-width: 2px;
  box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.35),inset 1px 1px 0px 1px ,inset -1px -1px 0 1px;
  box-sizing: border-box;
  display: inline-block;
  position: fixed;
  top: 0;
  right: 0;
  left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<header
  className="c0"
>
  test
</header>
`;
